zerver/views/user_groups.py

```diff
@@ -36,11 +38,26 @@
     user_groups_in_realm_serialized,
 )
 from zerver.lib.users import access_user_by_id, user_ids_to_users
-from zerver.lib.validator import check_bool, check_int, check_list
+from zerver.lib.validator import check_bool, check_dict_only, check_int, check_list, check_union
 from zerver.models import NamedUserGroup, UserProfile
 from zerver.models.users import get_system_bot
 from zerver.views.streams import compose_views
 
+check_group_setting: Validator[Union[int, AnonymousSettingGroupDict]] = check_union(
+    [
+        check_int,
+        cast(
+            Validator[AnonymousSettingGroupDict],
+            check_dict_only(
+                [
+                    ("direct_members", check_list(check_int)),
+                    ("direct_subgroups", check_list(check_int)),
+                ]
+            ),
+        ),
+    ]
+)
```

Comment:
I guess next time you're looking at writing a validator function like this, it's probably worth just converting the file to the `typed_endpoint` / Pydantic system first, since it'll save a migration later.